### DISCUSSION TITLE
Internal (or external?) error

### DISCUSSION THREAD

**Discussion by sfederici** [leopard-js/leopard-mentors#6] [Cleanup Crew]
### Scratch Project URL

https://scratch.mit.edu/projects/992215015/

### Leopard Project URL

https://leopardjs.com/projects/06O9Xowsoj

### What kind of error(s) are you experiencing?

- [ ] Leopard is failing to generate JavaScript code
- [ ] The JavaScript code generated by Leopard doesn't work correctly
- [ ] The JavaScript code generated by Leopard is confusing or needs improvement
- [X] Other

### Description of issue(s)

First time I clicked "see project page" and then "see inside" I got an "external error" (or internal? sorry, don't remember exactly)

**Comment by towerofnix** [comment]
Thanks! Would you be able to share what web browser and operating system you are using? I tried on Safari, Chrome, and Firefox, all on macOS, as well as with "throttled" network speed on Chrome (to simulate a slow connection), and wasn't able to get the same sort of issue (switching between project page, see inside, and back, starting from either side, didn't error).

**Reply by r4ph1-2015** [reply]
i used opera gx

**Reply by r4ph1-2015** [reply]
windows 11 newest version

**Reply by r4ph1-2015** [reply]
cooler master , RTX 5090

**Comment by sfederici** [comment]
As I said, this happened just once, the very first time I clicked "see project page" and then "see inside". I was using Firefox 127.0 (64 bit) on macOS Sonoma 14.3.1.

**Reply by towerofnix** [reply]
Thank you. We'll keep an eye out! I am not sure if @PullJosh ([away for a couple days](https://github.com/leopard-js/leopard/pull/199#issuecomment-2205894220)) automatically keeps a log of errors such as this. If you run into this again, it would be a help if you could open your browser console and take a screenshot of any error messages that appear to have shown up. I understand this might not be feasible e.g. if a student runs into the issue, so we'll hopefully have some word on how to help debug this sort of thing in the near future.

**Reply by PullJosh** [reply]
I do not currently log errors like this. Any recommendations for the best way to store that information? (Roll your own in the database, Sentry, etc)

**Reply by towerofnix** [reply]
No personal experience, sorry! Sentry seems very well-established albeit pricey if you're only getting light usage out of it — maybe the developer/free tier is OK, but you'd have to review that yourself.

Best to review privacy concerns also, i.e. we probably can't avoid occasionally including details like sprite names, but we can (and should) filter out errors that are flat-out thrown by the project code itself. We're interested in editor and website issues, not Leopard runtime errors (that atm can't be distinguished from user code w/ unhandled exceptions).

I don't know if it's standard or necessary to include info about logged errors in the Privacy Policy, but that's relevant to consider, too.

**Reply by r4ph1-2015** [reply]
I have the same problem
